AbilityChanger/Abilities/DashSlash.cs

```diff
@@ -12,16 +12,89 @@ public class DashSlash : AbilityManager {
         public override void OnFsmEnable(On.PlayMakerFSM.orig_OnEnable orig, PlayMakerFSM self)
         {
             orig(self);
+            if (self.gameObject.name == "Dash Slash" && self.FsmName == "damages_enemy")
+            {
+                Log("oi");
+                #region Contact
+                /*self.Intercept(new TransitionInterceptor()
+                {
+                    fromState = "Grandparent",
+                    toStateDefault = "Idle",
+                    eventName = "FINISHED",
+                    toStateCustom = "Idle",
+                    shouldIntercept = () => currentAbility.hasContact(),
+                    onIntercept = (fsmstate, fsmevent) => HandleContact(self.FsmVariables.FindFsmGameObject("Collider").Value)
+
+                });*/
+                self.InsertAction("Send Event", new CustomFsmAction(() => Log("ola")), 9);// HandleContact(self.FsmVariables.FindFsmGameObject("Collider").Value)), 5);
+
+                self.InsertAction("Parent", new CustomFsmAction(() => Log("ola1")), 5);// HandleContact(self.FsmVariables.FindFsmGameObject("Collider").Value)), 5);
+                self.InsertAction("Grandparent", new CustomFsmAction(() => Log("ola2")), 5);// HandleContact(self.FsmVariables.FindFsmGameObject("Collider").Value)), 5);
+
+                #endregion
+            }
+
+
+
             if (self.gameObject.name == "Knight" && self.FsmName == "Nail Arts")
             {
-                self.Intercept(new TransitionInterceptor(){
-                    fromState ="Dash Slash Ready",
-                    eventName ="DASH END",
-                    toStateDefault="DSlash Start",
-                    toStateCustom="Regain Control",
-                    shouldIntercept = () => this.hasTrigger(),
-                    onIntercept = (fsmstate,fsmevent) => this.handleAbilityUse(fsmstate,fsmevent)
-                });
+                #region Start
+                /*self.Intercept(new TransitionInterceptor()
+                {
+                    fromState = "Dash Slash Ready",
+                    eventName = "DASH END",
+                    toStateDefault = "DSlash Start",
+                    toStateCustom = "DSlash Start",
+                    shouldIntercept = () => currentAbility.hasStart(),
+                    onIntercept = (fsmstate, fsmevent) => HandleStart()
+                });*/
+
+                self.AddAction("Dash Slash",new CustomFsmAction(()=> HandleStart()));
```

Comment:
do we not want to skip the existing vanilla states? 

also could just use AddCustomAction directly here